### PR TITLE
Align bridge with kubetron

### PR DESCRIPTION
**Pull request by yuvalif**
This PR addresses issue #33 . But, actually did **not** align with kubetron implementation....
(1) I prefer the channel approach to get the pod over polling the k8s file - seems cleaner
(2) I prefer a programmatic approach for managing the interfaces over a scripted approach

### CODE REVIEW

File: pkg/network/bridge/plugin.go [hunk 1]
```diff
@@ -205,30 +206,94 @@ func (nbdp *NetworkBridgeDevicePlugin) attachPods() {
...
+		PeerName: nicName}
 
 	err = netlink.LinkAdd(link)
 	if err != nil {
 		return err
 	}
 
+	// set interface up
+	err = netlink.LinkSetUp(link)
+	if err!= nil {
```
Comment (phoracek) [inline pkg/network/bridge/plugin.go]: please run gofmt on the document

File: pkg/network/bridge/plugin.go [hunk 2]
```diff
@@ -205,30 +206,94 @@ func (nbdp *NetworkBridgeDevicePlugin) attachPods() {
...
+	// set peer interface up
+	err = netlink.LinkSetUp(peer)
+	if err!= nil {
+		netlink.LinkDel(link)
+		return err
+	}
+
+	// set back to the original namespace
+	netns.Set(originalNS)
+	// TODO: do we need that? Is error handling needed here?
```
Comment (phoracek) [inline pkg/network/bridge/plugin.go]: we should investigate it first, or just try to fail one Allocate call so it does not get out to the original netns and then try to add a new pod to bridge

**Review by phoracek — CHANGES_REQUESTED** [review]
Looks good, just a few comments.

File: pkg/network/bridge/plugin.go (hunk 1 above)
Comment (yuvalif) [reply pkg/network/bridge/plugin.go]: strange, go fmt run as part of the makefile. anyway, fixed now.

**Comment by yuvalif** [comment]
[[epic-network-resource] Learn about device plugins and Kubetron](https://trello.com/c/xoqcl0bq/94-epic-network-resource-learn-about-device-plugins-and-kubetron)

File: pkg/network/bridge/plugin.go (hunk 2 above)
Comment (yuvalif) [reply pkg/network/bridge/plugin.go]: setting back is mandatory. if not, the next pod that needs a device would fail

File: Gopkg.lock [hunk 3]
```diff
@@ -94,7 +94,7 @@
   branch = "master"
   name = "github.com/kubevirt/device-plugin-manager"
   packages = ["pkg/dpm"]
-  revision = "b70ea7a8e3a534ecb00f694e58894e1c0d3f3653"
+  revision = "a2dafa739e0340c49d0fdcc2aa80dfa68edcea0f"
```
Comment (yuvalif) [inline Gopkg.lock:97]: this is the version of DPM with the race condition fix

File: Gopkg.toml [hunk 4]
```diff
@@ -34,6 +34,6 @@
   name = "github.com/kubevirt/device-plugin-manager"
 
 [[constraint]]
-  version = "1.10"
+  version = "~v1.10"
```
Comment (yuvalif) [inline Gopkg.toml:37]: prevent go dep from taking k8s 11

**Review by phoracek — APPROVED** [review]

**Pull request merged**